tests/unit/modules/network/sonic/fixtures/sonic_bgp.yaml

```diff
@@ -0,0 +1,143 @@
+---
+merged_01:
+  module_args:
+    config:
+      - bgp_as: 4
+        router_id: 10.2.2.4
+        log_neighbor_changes: False
+        timers:
+          holdtime: 20
+          keepalive_interval: 30
+        bestpath:
+          as_path:
+            confed: True
+            ignore: True
+            multipath_relax: False
+            multipath_relax_as_set: True
+          compare_routerid: True
+          med:
+            confed: True
+            missing_as_worst: True
+            always_compare_med: True
+        max_med:
+          on_startup:
+            timer: 667
+            med_val: 7878
+  existing_bgp_config:
+    - path: "/data/openconfig-network-instance:network-instances/network-instance=default/protocols/protocol=BGP,bgp/bgp/global"
+      response:
+        code: 200
+        value:
+          openconfig-network-instance:global:
+            config:
+              as: 4
+    - path: "data/sonic-vrf:sonic-vrf/VRF/VRF_LIST"
+      response:
+        code: 200
+        value:
+          sonic-vrf:VRF_LIST:
+            - vrf_name: default
+  expected_config_requests:
+    - path: "/data/openconfig-network-instance:network-instances/network-instance=default/protocols/protocol=BGP,bgp/bgp/global/config"
+      method: "patch"
+      data:
+        openconfig-network-instance:config:
+          router-id: "10.2.2.4"
+          as: 4.0
+    - path: "/data/openconfig-network-instance:network-instances/network-instance=default/protocols/protocol=BGP,bgp/bgp/global/config/hold-time"
+      method: "patch"
+      data:
+        hold-time: '20'
+    - path: "/data/openconfig-network-instance:network-instances/network-instance=default/protocols/protocol=BGP,bgp/bgp/global/config/keepalive-interval"
+      method: "patch"
+      data:
+        keepalive-interval: '30'
+    - path: "/data/openconfig-network-instance:network-instances/network-instance=default/protocols/protocol=BGP,bgp/bgp/global/max-med"
+      method: "patch"
+      data:
+        max-med:
+          config:
+            max-med-val: 7878
+            time: 667
+    - path: "/data/openconfig-network-instance:network-instances/network-instance=default/protocols/protocol=BGP,bgp/bgp/global/route-selection-options"
+      method: "patch"
+      data:
+        route-selection-options:
+          config:
+            external-compare-router-id: True
+            compare-confed-as-path: True
+            ignore-as-path-length: True
+            med-confed: True
+            med-missing-as-worst: True
+            always-compare-med: True
+    - path: "/data/openconfig-network-instance:network-instances/network-instance=default/protocols/protocol=BGP,bgp/bgp/global/use-multiple-paths/ebgp/config"
+      method: "patch"
+      data:
+        openconfig-network-instance:config:
+          as-set: True
+deleted_01:
+  module_args:
+    config:
```

Comment:
There are several attributes for which deletion is not being tested here. Is there a reason not to test deletion for these attributes?

- timers
- max_med
- bestpath/as_path/multipath_relax_as_set
- med/always_compare_med

Reply:
Thanks for adding these attributes to the test.

Resolved.